schema/scope.go

```diff
@@ -92,10 +92,28 @@ func (s *ScopeSchema) Serialize(data any) (any, error) {
 	return s.ObjectsValue[s.RootValue].Serialize(data)
 }
 
-func (s *ScopeSchema) ApplyScope(_ Scope) {
+func (s *ScopeSchema) ApplyScope(scope Scope, namespace string) {
+	if namespace == DEFAULT_NAMESPACE {
+		// Same namespace, so pass in self with default namespace.
+		for _, v := range s.ObjectsValue {
+			v.ApplyScope(s, DEFAULT_NAMESPACE)
+		}
+	} else {
+		// Separate namespace. Pass in the other scope instead.
+		for _, v := range s.ObjectsValue {
+			v.ApplyScope(scope, namespace)
+		}
+	}
```

Comment:
I missed the fact that the nested calls to `ApplyScope()` had different arguments.

Yes, I think it would be better to have a local variable set to the appropriate argument value and then have a single, common invocation of `ApplyScope()`.

However, I don't really understand what is going on here.  (In none of the other `ApplyScope()` calls so far did we need to select which value to pass in the first argument.)  What we're doing is applying a scope to a scope?  When does/can that actually happen?

Reply:
When a namespace isn't specified, the scopes just apply themselves, because the default scope should be the scope the objects are a part of.

Reply:
Can you give me an example of when a scope or namespace would be applied to a _scope_?

Reply:
So the existing case is every scope applies itself recursively to allow references to resolve.
The new case is to retain the prior behavior with the default namespace, but also allow referencing a scope from another step. Here is an example in a workflow: https://github.com/jaredoconnell/perfconf-2024-demos/blob/394bb0462216f61e6f483aad7afa89281d858d50/sysbench-cpu-workflow.yaml#L15